packages/godcap/bonsai/client.go

```diff
@@ -120,37 +129,44 @@ func (c *Client) s3(method string, url string, body io.Reader) ([]byte, error) {
 	return httpBody, nil
 }
 
-func (c *Client) api(method string, path string, body io.Reader, response interface{}) error {
+func (c *Client) api(method string, path string, body io.Reader, response interface{}) (int, error) {
```

Comment:
Is there necessary to return the statusCode? because the `UploadImage` only need to check if the statusCode is success, which it's already done in the `api` call.

Reply:
Yeah, now adding a status code to `api` makes sense. I think we can keep the ELF file, maybe move it to the `zkdcap` package? 
And let's skip computing an image ID; it seems too complex. Users can decide what they want to upload themselves, since this `bonsai` package is meant to be a general SDK.

Reply:
https://github.com/automata-network/dcap-sdk/pull/4/commits/28ef9b8abc49bfeb1da69fb4a56599b83488de66

Moved the ELF to `zkdcap`